changelog-extractor.py

```diff
@@ -0,0 +1,144 @@
+"""Script parses changelog from git based on 2 latest tags, formates changelog to .md format, creates/updates release on gitlab, posts to slack
+"""
+import subprocess
+import os
+
+def getMdFormattedChangelog(newCommit, oldCommit) : 
+    #saving parameters this way to make sure that they are read properly on bash command
+    get_log_format = '%x1f'.join(['%b', '%h']) + '%x1e'
+    bashCommand = 'git --no-pager --git-dir=.git log --first-parent --format="%s"' % get_log_format   + " "+ newCommit+"..." + oldCommit
+    #getting raw changelog from git
+    rawChangelog = sendCommand(bashCommand).replace("\"", "")
+    return  ConvertChangelogTextToMd(rawChangelog, None)
+
+def buildCommandForTagNotes(reqType, cleanChangeLog, tag) : 
+    formattedProjectPath = os.environ['CI_PROJECT_PATH'].replace("/","%2F")  
+    return 'curl  -X '+reqType+'  --header "PRIVATE-TOKEN: $GITLAB_API_PRIVATE_TOKEN" -d description="'+cleanChangeLog+ '" https://git.smarpsocial.com/api/v4/projects/"'+ formattedProjectPath+ '"/repository/tags/"'+tag+'"/release'
```

Comment:
please extract repository name to env variable

Reply:
what for?

Reply:
are you talking about https://git.smarpsocial.com/api/v4/projects/ ? then it could be done, ys